scimma_admin/hopskotch_auth/views.py

```diff
@@ -26,6 +26,15 @@ def redirect_with_error(request, operation, reason, redirect_to):
     return redirect(redirect_to)
 
 
+def client_ip(request):
+    """Determine the original client IP address, taking into account headers set
+    by the load balancer, if they exist.
+    """
+    if "HTTP_X_FORWARDED_FOR" in request.META:
+        return request.META["HTTP_X_FORWARDED_FOR"]
```

Comment:
I can make that stylistic change, I hadn't realized that was an option from my reading of the Django docs. 

I had also not noticed that ELB would append to the header rather than replacing it. We trust ELB to tell us the truth, so it seems like we should keep the rightmost address if there is a list, but this will break down if for some reason multiple layers of proxying are going on. Alternatively we could keep everything, as it is now, and document that anyone doing security analysis must treat addresses further to the left in list with increasing suspicion, as they might not have come from trustworthy sources?

Reply:
After discussing with Ron, in the case that the forwarding header is a comma separated list I am returning a more elaborate string which singles out the last item in the list as being trustworthy, also contains the full header in case it turns out to be of any value.